### PR TITLE
WIP: sympify: add kwarg for how floating-point literals are handled

### PR DESCRIPTION
**Pull request by cbm755**
This is just for discussion, do not merge this!

See Issue #13425.

### CODE REVIEW

File: sympy/parsing/sympy_parser.py [hunk 1]
```diff
@@ -848,7 +848,11 @@ def stringify_expr(s, local_dict, global_dict, transformations):
         tokens.append((toknum, tokval))
 
     for transform in transformations:
-        tokens = transform(tokens, local_dict, global_dict)
+        # TODO: yuck!
```
Comment (asmeurer) [inline sympy/parsing/sympy_parser.py]: That's why I suggested to make a second transformer that converts `Float(string)` into `Float(string, precision)`.

Comment (cbm755) [reply sympy/parsing/sympy_parser.py]: That makes sense!  Sorry I didn't understand your original comment (I did this without really understanding any of this code in fact ;-)

File: sympy/core/sympify.py [hunk 2]
```diff
@@ -355,8 +369,16 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
...
+    from sympy.parsing.sympy_parser import auto_number as t_auto_number
 
     transformations = standard_transformations
+    if default_float_dps is not None:
+        myfcn = lambda x,y,z: t_auto_number(x,y,z,default_float_dps=default_float_dps)
+        # TODO: this doesn't seem to work, got:
+        #   ValueError: could not convert string to float: '0.1'
+        #transformations += (myfcn,)
+        # So instead, something horrible
+        transformations = tuple(myfcn if x is t_auto_number else x for x in transformations)
```
Comment (cbm755) [inline sympy/core/sympify.py]: @asmeurer I still need a way to get `default_float_dps` to the transformation, whether I've written a new one or not...  I don't see how to do that without mucking around in the `standard_transformations` tuple.

Perhaps we could pass a "transf_config" dict around to all the transformations?  Like `local_dict` and `global_dict` currently?

File: sympy/core/sympify.py [hunk 3]
```diff
@@ -256,6 +256,20 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     >>> kernS(s)
     -2*(-(-x + 1/x)/(x*(x - 1/x)**2) - 1/(x*(x - 1/x))) - 1
 
+    Floating point literals ... TODO: explain this (!)  Consider:
+
+    >>> sympify('0.1*1.4142135623730950488016887242097')**2
+    0.020000000000000002220446049250313
```
Comment (asmeurer) [inline sympy/core/sympify.py]: I think your original example of `2**0.5` is better. The behavior in this case is basically correct (a product of floats should use the minimum of the precisions of the arguments), and it seems odd that you would want to change it.

**Comment by asmeurer** [comment]
Just realized this whole thing isn't needed if you just pass the `rational=True` flag and then evalf to the desired precision. It isn't actually the same for `sympify` because the whole expression would get evaluated, but for the original `N('2**0.5', 30)` it would work just fine.

**Comment by asmeurer** [comment]
I'm not opposed to adding the flag to the `auto_number` transformer, and then someone can use `functools.partial` to customize it with `parse_expr` if they want. But I'm weary on adding more flags to `sympify`. I think only the most common options should be on sympify and custom parsing stuff should be left to `parse_expr`.

**Comment by czgdp1807** [comment]
Closing this in favour of #16963

**Pull request closed without merging**